pyroomacoustics/utilities.py

```diff
@@ -462,10 +544,64 @@ def levinson(r, b):
         epsilon = epsilon*(1 - np.abs(gamma)**2)
         delta = np.dot(np.conj(r[1:j+1]), np.flipud(x))
         q = (b[j, ] - delta)/epsilon
-        x = np.concatenate((x, np.zeros(1 if len(b.shape) == 1 else (1, b.shape[1]))), axis=0) + q*np.conj(a[::-1, np.newaxis])
+        if len(b.shape) == 1:
+            x = np.concatenate((x, np.zeros(1))) + q * np.conj(a[::-1])
+        else:
+            x = np.concatenate((x, np.zeros((1, b.shape[1]))), axis=0) + q * np.conj(a[::-1, np.newaxis])
 
     return x
 
+
+def bac(x, p):
```

Comment:
This implementation is quite inefficient. This can be implemented efficiently  using the FFT.

1. Take FFT with appropriate zero-padding
2. Take squared magnitude
3. Take iFFT
4. Apply appropriate normalization and truncation

Reply:
I guess the difference in efficiency might depend on the length of x, `L`. If `p < log_2(L)`, then the direct sum might be more efficient (on paper at least, the python loop will be super slow).

Reply:
Here is an example of the cross-correlation implemented in this way: https://github.com/LCAV/pyroomacoustics/blob/master/pyroomacoustics/sync.py#L45

Reply:
At least compute the inner loop with `np.inner` or `np.mean`